### PR TITLE
Implement NavigationBar

### PR DESCRIPTION
**Pull request by vasily-kirichenko**
- [x] `NavigationImpl` seems to give not enough results. It does not return module `let`s, for instance.
- [x] It's unclear how to implement `NavigateToItem` method (it does not work at all at the moment), and how to do it Roslyn idiomatically. 
- [x] Make it work on signature files

### CODE REVIEW

**Comment by cartermp** [comment]
*yuusss*

**Comment by vasily-kirichenko** [comment]
It's ready

image

File: vsintegration/src/FSharp.Editor/NavigationBarItemService.fs [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+                let workspace = document.Project.Solution.Workspace
+                let navigationService = workspace.Services.GetService<IDocumentNavigationService>()
+                
+                if navigationService.CanNavigateToPosition(workspace, document.Id, span.Start) then
+                    navigationService.TryNavigateToPosition(workspace, document.Id, span.Start) |> ignore
+                else
+                    let notificationService = workspace.Services.GetService<INotificationService>()
+                    notificationService.SendNotification(EditorFeaturesResources.The_definition_of_the_object_is_hidden, severity = NotificationSeverity.Error)
+
+                //view.Selection.Select(SnapshotSpan(view.TextBuffer.CurrentSnapshot, Span(span.Start, span.Length)), false)
```
Comment (forki) [inline vsintegration/src/FSharp.Editor/NavigationBarItemService.fs]: can you please remove that comment?

**Comment by vasily-kirichenko** [comment]
It's ready.

**Comment by KevinRansom** [comment]
@dotnet-bot test this please

**Comment by KevinRansom** [comment]
@vasily-kirichenko 
Thank you for taking care of this.

Kevin

**Pull request merged**